diff --git a/node_modules/react-native-sodium/android/build.gradle b/node_modules/react-native-sodium/android/build.gradle
index 1f5e62f..416578e 100644
--- a/node_modules/react-native-sodium/android/build.gradle
+++ b/node_modules/react-native-sodium/android/build.gradle
@@ -11,12 +11,13 @@ buildscript {
 apply plugin: 'com.android.library'

 android {
-    compileSdkVersion 28
-    buildToolsVersion "28.0.3"
+    compileSdkVersion 35
+    buildToolsVersion "35.0.0"
+    ndkVersion rootProject.ext.ndkVersion
     defaultConfig {
-        minSdkVersion 16
-        compileSdkVersion 28
-        targetSdkVersion 28
+        minSdkVersion 21
+        compileSdkVersion 35
+        targetSdkVersion 34
         versionCode 1
         versionName "0.1"
         compileOptions {
@@ -25,7 +29,15 @@ android {
         }
         externalNativeBuild {
             cmake {
-                arguments "-DANDROID_TOOLCHAIN=clang"
+                arguments (
+                    "-DANDROID_TOOLCHAIN=clang",
+                    "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
+                    "-DANDROID_NATIVE_API_LEVEL=21",
+                    "-DANDROID_ARM_MODE=arm",
+                    "-DCMAKE_C_FLAGS=-fno-strict-aliasing -fwrapv",
+                    "-DCMAKE_CXX_FLAGS=-fno-strict-aliasing -fwrapv",
+                    "-DCMAKE_BUILD_TYPE=Release"
+                )
             }
         }
         ndk {
diff --git a/node_modules/react-native-sodium/build.sh b/node_modules/react-native-sodium/build.sh
index eb61bc6..4740c85 100755
--- a/node_modules/react-native-sodium/build.sh
+++ b/node_modules/react-native-sodium/build.sh
@@ -39,6 +39,9 @@ do
       dist-build/android-arm.sh
       ;;
     "arm")
+      echo "*** enable 16 KB page-size support on Android 15/16+ ***"
+      export LDFLAGS="$LDFLAGS -Wl,-z,max-page-size=16384"
+
       dist-build/android-armv7-a.sh
       dist-build/android-armv8-a.sh
       ;;
diff --git a/node_modules/react-native-sodium/ios/RCTSodium/RCTSodium.m b/node_modules/react-native-sodium/ios/RCTSodium/RCTSodium.m
index 5599c58..e4fd8a8 100644
--- a/node_modules/react-native-sodium/ios/RCTSodium/RCTSodium.m
+++ b/node_modules/react-native-sodium/ios/RCTSodium/RCTSodium.m
@@ -5,8 +5,8 @@
 //  Created by Lyubomir Ivanov on 9/25/16.
 //  Copyright © 2016 Lyubomir Ivanov. All rights reserved.
 //
-#import "RCTBridgeModule.h"
-#import "RCTUtils.h"
+#import <React/RCTBridgeModule.h>
+#import <React/RCTUtils.h>
 #import "sodium.h"

 #import "RCTSodium.h"